sdk/web-pubsub/web-pubsub-express/src/utils.ts

```diff
@@ -56,8 +56,20 @@ export async function convertHttpToEvent(request: IncomingMessage): Promise<Mess
     }
   }
 
-  normalized.body = await readRequestBody(request);
-  return normalized;
+  const body = (normalized.body = await readRequestBody(request));
+  const receivedEvent = HTTP.toEvent(normalized);
+  if (isJsonData(receivedEvent)) {
+    // CloudEvent JSONParser wrap string with quotes however it is not the case for numbers or booleans
+    // https://github.com/cloudevents/sdk-javascript/blob/main/src/parsers.ts#L29
+    // We workaround it here by rewrite the parsed data property
+    receivedEvent.data = JSON.parse(body);
+  }
+
+  return receivedEvent;
+}
+
+export function isJsonData(event: CloudEvent): boolean {
+  return event.datacontenttype?.startsWith("application/json;") === true ? true : false;
```

Comment:
Can shorten this slightly if you are just trying to avoid returning `undefined`
```suggestion
  return Boolean(event.datacontenttype?.startsWith("application/json;"));
```